services/cd-service/pkg/cmd/server.go

```diff
@@ -125,8 +126,17 @@ func RunServer() {
 			logger.FromContext(ctx).Fatal("azure.auth.error: pgpKeyRing is required to authenticate manifests when \"KUBERPULT_AZURE_ENABLE_AUTH\" is true")
 		}
 
+		var reader auth.GrpcContextReader
+		if c.DexMock {
+			c.DexEnabled = true
```

Comment:
rather than changing a configuration silently, please return an error if they configs do not match:
```suggestion
		if c.DexMock {
		        if !c.DexEnabled {
		            return nil, errors.New("dexEnabled must be true if dexMock is true")
		        }
```